pkg/console/controllers/service/controller.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/openshift/console-operator/pkg/console/status"
+	"github.com/openshift/console-operator/pkg/console/subresource/service"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+const (
+	// key is basically irrelevant
+	controllerWorkQueueKey = "service-sync-work-queue-key"
+	controllerName         = "ConsoleServiceSyncController"
+)
+
+// ctrl just needs the clients so it can make requests
+// the informers will automatically notify it of changes
+// and kick the sync loop
+type ServiceSyncController struct {
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	// live clients, we dont need listers w/caches
+	serviceClient coreclientv1.ServicesGetter
+	// names
+	targetNamespace string
+	serviceName     string
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+// factory func needs clients and informers
+// informers to start them up, clients to pass
+func NewServiceSyncController(
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	corev1Client coreclientv1.CoreV1Interface,
+	serviceInformer coreinformersv1.ServiceInformer,
+	// names
+	targetNamespace string,
+	serviceName string,
+	// events
+	recorder events.Recorder,
+) *ServiceSyncController {
+
+	corev1Client.Services(targetNamespace)
+
+	ctrl := &ServiceSyncController{
+		operatorConfigClient: operatorConfigClient,
+		serviceClient:        corev1Client,
+		// names
+		targetNamespace: targetNamespace,
+		serviceName:     serviceName,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	serviceInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		serviceInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *ServiceSyncController) sync() error {
+	startTime := time.Now()
+	klog.V(4).Infof("started syncing service %q (%v)", startTime)
+	defer klog.V(4).Infof("finished syncing service %q (%v)", time.Since(startTime))
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorsv1.Managed:
+		klog.V(4).Infoln("console is in an managed state: syncing service")
+	case operatorsv1.Unmanaged:
+		klog.V(4).Infoln("console is in an unmanaged state: skipping service sync")
+		return nil
+	case operatorsv1.Removed:
+		klog.V(4).Infoln("console is in an removed state: deleting service")
+		return c.removeService()
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	if !management.IsOperatorManaged(operatorConfig.Spec.ManagementState) {
```

Comment:
this is not needed anymore since we have `switch`

Reply:
Ah, thats right, I was pondering this a bit more since this `func` does a few extra things, but I agree, we don't need.